src/modules/reservation.js

```diff
@@ -0,0 +1,117 @@
+                </div>
+
+                <h1 class="meal-title">${meal.strMeal}</h1>
+
+                <div class="item-specification-container">
+                    <ul class="item-specification">
+                        <li class="item">Area:${meal.strArea}</li>
+                        <li class="item">Category :${meal.strCategory}</li>
+                    </ul>
+                    <ul class="item-specification">
+                        <li class="item">Ingredient:${meal.strIngredient8}</li>
+                        <li class="item">Ingredient:${meal.strIngredient9}</li>
+                    </ul>
+                </div>
+            </div>
+
+            <div class="meal-container-order">
+                <h2 class="meal-reservation">Reservation History <span class="meal-reservation-span"></span></h2>
+
+                <div>
+                    <ul class="order-list-${meal.idMeal}">
+                    </ul>
+                </div>
+                
+                <h3 class="add-reservation">Add a reservation</h3>
+                <form class="order-form">
+                    <input type="text" class="form-control" id="userName" placeholder="Your name" required />
+                    <input type="date" class="form-control" id="start-date" placeholder="Start date" required />
+                    <input type="date" class="form-control" id="end-date" placeholder="End date" required />
+                    <input type="submit" class="btnReserve" id="btnReserve" value="Reserve" />
+                </form>
+
+            </div>
+         </div>
+        </div>`;
+
+  document.querySelector('.reservation-popup').innerHTML = content;
+
+  // Adding reservation history count
+  document.querySelector('.meal-reservation-span').innerHTML = reservetionInfoArray.length || 0;
+
+  // Adding reservation history to the modal
+  const reservationUl = document.querySelector(`.order-list-${meal.idMeal}`);
+  setReservationHistory(reservetionInfoArray, reservationUl);
+
+  // close Modal
+  document.querySelector('#closeBtn').addEventListener('click', () => {
+    document.querySelector('.Reservation-modal').style.display = 'none';
+  });
+
+  // Adding reservation
+  const reservationFrom = document.querySelector('.order-form');
+  reservationFrom.addEventListener('submit', (event) => {
+    event.preventDefault();
+    const userName = document.querySelector('#userName').value;
+    const startDate = document.querySelector('#start-date').value;
+    const endDate = document.querySelector('#end-date').value;
+    addReservation(meal.idMeal, userName, startDate, endDate);
+    reservationFrom.reset();
+
+    let updatedReservetionInfoArray = [];
+
+    setTimeout(() => {
+      (async () => {
+        try {
+          updatedReservetionInfoArray = await getReservation(meal.idMeal);
+        } catch (error) {
+          /* eslint-disable */
+          console.error(error.message);
+          /* eslint-enable */
```

Comment:
please suggest what could be an alternative here to get the message in case an error occurs? [in this particular case inside a catch block] 

Or, should we remove it any ways? Thanks

Reply:
- [ ] If the user needs to see the error message, you can display it on the webpage, style it and reword it so that it's not technical and a user can understand what the problem is and what they should do. If the user does not need to see it, you should handle it in code so that the application does not crash. logging the error to the console can leak confidential information to hackers and users do not always check the console when using web apps. 😄